moveit_core/collision_detection_fcl/src/collision_common.cpp

```diff
@@ -495,13 +494,19 @@ bool distanceCallback(fcl::CollisionObjectd* o1, fcl::CollisionObjectd* o2, void
 
   double dist_threshold = cdata->req->distance_threshold;
 
-  const std::pair<std::string, std::string>& pc = cd1->getID() < cd2->getID() ?
-                                                      std::make_pair(cd1->getID(), cd2->getID()) :
-                                                      std::make_pair(cd2->getID(), cd1->getID());
+  thread_local std::pair<std::string, std::string> pc = std::make_pair(cd1->getID(), cd2->getID());
+  pc = cd1->getID() < cd2->getID() ? std::make_pair(cd1->getID(), cd2->getID()) :
+                                     std::make_pair(cd2->getID(), cd1->getID());
```

Comment:
Can we completely get rid of heap allocation for `std::string` here? See https://github.com/AndyZe/moveit/pull/2

Reply:
Merged. Thanks. What you did with std::string& to avoid allocation is pretty clever. It seems not to really help runtime over the `thread_local` version but I guess it has the benefit of being more readable.